src/Controller/Base.php

```diff
@@ -155,10 +155,16 @@ protected function redirect($url, $status = 302)
      * @param array  $parameters An array of parameters
      * @param int    $status     The status code to use for the Response
      *
-     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|Symfony\Component\HttpFoundation\JsonResponse
      */
     protected function redirectToRoute($route, array $parameters = [], $status = 302)
     {
+        // Don't redirect on ajaxy requests, but abort with a 500 code instead.
+        if ($this->app['request']->isXmlHttpRequest()) {
+            $response = ['error' => ['message' => 'Redirect detected. Check if you are still logged on.'] ];
+            return new JsonResponse($response, 500);
+        }
```

Comment:
Currently, all 'ajaxy' requests that will redirect will fail silently. What happens is that it gets redirected, and returns an HTML page of the login screen, with a `200` status. As far as I can tell these just fail silently, and in javascript the `.done()` or `.success()` will be triggered, which is _not_ correct. 

This change will send a 500, so in the requesting javascript it can be caught. At the very least it'll trigger the `.fail()` method, allowing us to give feedback to the user.